### PR TITLE
P4 3149 housekeeping

### PR DESCRIPTION
**Pull request by matt-crowson-moj**
**Changes:**

Housekeeping exercise to keep the project up to date with latest 3rd party libraries to keep on top of their improvements and fixes.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -9,14 +9,17 @@ jobs:
     executor:
       name: hmpps/java
       tag: "11.0"
+    environment:
+      _JAVA_OPTIONS: -Xmx512m -XX:ParallelGCThreads=2 -XX:ConcGCThreads=2 -Djava.util.concurrent.ForkJoinPool.common.parallelism=2 -Dorg.gradle.daemon=false
```
Comment (matt-crowson-moj) [inline .circleci/config.yml:13]: This improves the memory management when running the tests.  I was starting to see out of memory errors.

File: .circleci/config.yml [hunk 2]
```diff
@@ -9,14 +9,17 @@ jobs:
...
+
     steps:
       - checkout
       - restore_cache:
           keys:
             - gradle-{{ checksum "build.gradle.kts" }}
             - gradle-
       - run:
-          command: ./gradlew -Dorg.gradle.jvmargs="--illegal-access=permit" -Dkotlin.daemon.jvm.options="--illegal-access=permit" build
```
Comment (matt-crowson-moj) [inline .circleci/config.yml:19]: No longer needed.

File: build.gradle.kts [hunk 3]
```diff
@@ -70,6 +69,11 @@ tasks {
     exclude("**/integration/*", "**/migrations/*")
   }
 
+  withType<JavaCompile> {
```
Comment (matt-crowson-moj) [inline build.gradle.kts:72]: This resolves a compiler warning that started showing as part of the update.

File: src/main/resources/db/migration/ddl/V1_0__baseline.sql [hunk 4]
```diff
@@ -126,3 +126,28 @@ create table shedlock
 create index EVENTS_EVENTABLE_ID_IDX on events (eventable_id);
 
 create index JOURNEYS_MOVE_ID_IDX on journeys (move_id);
+
+CREATE TABLE SPRING_SESSION
```
Comment (matt-crowson-moj) [inline src/main/resources/db/migration/ddl/V1_0__baseline.sql:130]: So I had to bring this into here as part of the update to latest Spring Boot.  It was not playing nicely with the Flyway DB migration tool.  In short all of the DB tables needed are now included in the baseline script.

File: src/test/resources/application-test.yml [hunk 5]
```diff
@@ -23,6 +23,10 @@ spring:
       resourceserver:
         jwt:
           issuer-uri:
+  session:
+    store-type: jdbc
+    jdbc:
+      initialize-schema: embedded
```
Comment (matt-crowson-moj) [inline src/test/resources/application-test.yml:29]: The unit tests do not run DB migrations so need to tell it to run in memory (which is what the above is doing).

**Review by thomasleese — APPROVED** [review]

**Pull request merged**